src/server/transaction.cc

```diff
@@ -1194,6 +1200,32 @@ bool Transaction::NotifySuspended(TxId committed_txid, ShardId sid) {
   return false;
 }
 
+void Transaction::LogJournalOnShard(EngineShard* shard) {
+  // TODO: For now, we ignore non shard coordination.
+  if (shard == nullptr)
+    return;
+
+  if ((cid_->opt_mask() & CO::WRITE) == 0)
+    return;
```

Comment:
Maybe we should also skip blocking, because I didn't test them properly. Not sure they'll work out of the box